core/src/main/java/com/linecorp/armeria/server/auth/Authorizer.java

```diff
@@ -43,38 +43,28 @@
      */
     CompletionStage<Boolean> authorize(ServiceRequestContext ctx, T data);
 
+    /**
+     * Authorizes the given {@code data}.
+     *
+     * @return a {@link CompletionStage} that will resolve to {@link AuthorizationStatus}. If the future
+     *     resolves exceptionally, the request will not be authorized.
+     */
+    default CompletionStage<AuthorizationStatus> authorizeAndSupplyHandlers(ServiceRequestContext ctx,
+                                                                            @Nullable T data) {
+        if (data == null) {
```

Comment:
Could you explain what `null` data means and what the expected behavior of the method in Javadoc?

Reply:
NULL data means that the authorization material might be absent on the request (token missing or something).
The original code looked like the following:
```
        final Authorizer<HttpRequest> requestAuthorizer = (ctx, req) -> {
            final T token = tokenExtractor.apply(req.headers());
            if (token == null) {
                return CompletableFuture.completedFuture(false);
            }
            return authorizer.authorize(ctx, token);
        };
```
Now, it's been extracted into a separate class `DelegatingHttpRequestAuthorizer` and the authorization decision consolidated in the actual `Authorizer`, instead of some inline code somewhere inside the builder.
The `Authorizer` signature should have allowed NULL, but unfortunately it did not. So I fully preserve previous contract:
```
CompletionStage<Boolean> authorize(ServiceRequestContext ctx, T data);
```
Added following comment:
```
     * @param data an actual authorization data, like {@link HttpRequest}, token extracted from it or
     *     {@code null} if such authorization data is missing.
```

Reply:
Thanks a lot. Makes sense.